### ISSUE TITLE
Nerfing weld times

### ISSUE THREAD

**Issue by woothie** [HeavensGate/Eternal#19]
Extend the time required for most if not all welding interactions, both constructing and deconstructing. Longer durations to weld doors shut, break through walls, etc. Basically, stronger fortifications with longer durations.

--- Merged from NSS-Eternal

**Comment by Davnik** [comment]
Working on this would seem out of place and arbitrary... it would give players an option to fortify themselves though, if nothing else, and prevent players from yakkity saxing with doorway stopgaps.

**Comment by nullbear** [comment]
What's the for and against on accumulative welding time?

For example, lets say a wall requires 60 ticks to deconstruct. If you spend 15 ticks, and are interupted, it only requires 45 more ticks to finish cutting it if you went back to it before a certain time has passed. This could allow for longer welding/unwelding times, but reduce the tedium of having to start from the beginning if you're interupted. It could also allow for two people to work on the same job, in order to speed it up.

This could potentially have an inverse effect, where you don't NEED to weld something for the full time, but doing so enhances the effect. Such as quickly welding a door in one second to stop someone getting through right away, and welding a door for 10 seconds to delay entry for a bit longer.

Not really a necessary thing, just thoughts. It could be pretty easy to implement mutual welding tasks without the acummulative welding time too.

**Comment by staticcharged** [comment]
> For example, lets say a wall requires 60 ticks to deconstruct. If you spend 15 ticks, and are interupted, it only requires 45 more ticks to finish cutting it if you went back to it before a certain time has passed. This could allow for longer welding/unwelding times, but reduce the tedium of having to start from the beginning if you're interupted. It could also allow for two people to work on the same job, in order to speed it up.

I like the cumulative time idea, this makes a lot of sense. However I feel there would be no point in providing an option for an immediate or cumulative, slightly increased welding times because the option for an immediate weld would be overused and abused. 
If we were to provide the opportunity to weld a door in a second it would be at the cost of draining the welder- in my opinion.

**Comment by nullbear** [comment]
Yeah, i dont think it should be instant. But rather time-based and tiered.

For example it takes 1 second to weld the door, but if you weld it for longer, the weld is stronger, and takes longer to break. Maybe make the tier one weld breakable by spending some time on the door with a crowbar? second weld MUST be cut with a welder, third weld must be cut with a welder and take a substantial amount of time to cut.

**Comment by nullbear** [comment]
I think i'll start working on this, but i'd also like to work on something else related to this alongside it in the same commit, and i'd like some feedback from whoever's got authority her before i dedicate myself to it.

Essentially, im changing the way welding works a bit to be point based, rather than time based. This would effectively allow for more than one person to work on a wall at the same time, and for different levels of 'welding tools' to be added if ever needed. (at the moment, structures are hardcoded to use a certain amount of time, so its a lot of extra work to make 'faster' welders.) It will still be essentially the same, and it mostly just improves code reuseability. I may also nerf welders slightly. 

(essentially welders will be controlling the speed. And the task is constant. Ie. a standard welder will weld at 10 points per second. While a better welder could work at 20 points per second. The fastest weld would never be shorter than one full second of welding.)

As well, what i'd like some approval on, is a moderate change to code organisation, which will reduce item paths, attackby proc tedium, and typically just make the code a bit cleaner and easier to work with. Is to add a 'tag' var to objects.

How this relates to this issue:

Instead of

attackby(w, mob)
    if w == wrench
     do stuff
    if w == welder
      do stuff
etc.

it becomes

attackby(w, mob)
   if (wrench) in w.tags
     do stuff

where there is no notable change, but it removes the hardcoding of construction based on the items hard coded type, into what the item 'says' it is. This probably wont mean much unless you have multipurpose items, such as a knife with a wrench on the end or a leatherman or something.

The tags for a wrench, however, would probably look something like this:
tags = list(tool, wrench, material_steel)

material steel, could be used as a reference, to check if an object is conductive, meltable, and typically just a label for other stuff to use. For example if you want to make an acid that can melt steel, but nothing else, you could have it use this. Or for grinding something up, the material that you get back. Or whatever. idc. You don't need it it's just there as an example and for convenience.

wrench, would be used by structures and equipment to determine the type of tool it is, without having to hardcode to the object type, allowing for crosstype code reuseability, increasing organisation and reducing labor.

tool, could be used by mobs, for example by clawed races and monkeys and folks with brain damage, to determine if it is able to use the tool. Whereas something tagged as 'advancedtool' wouldnt be useable by them, 'tool' and 'simpletool' could be. It just reduces hardcoding, and increases code reuseability for less labor and better organisation.
-------
